platform/darwin/src/MGLFeature.mm

```diff
@@ -188,10 +195,47 @@ - (id)attributeForKey:(NSString *)key {
 }
 
 - (NS_DICTIONARY_OF(NSString *, id) *)featureDictionary {
-    // TODO: MGLShapeCollectionFeature
-    return nil;
+    
+    return @{@"type":@"Feature",
+             @"properties":(self.attributes) ? self.attributes : [NSDictionary dictionary],
+             @"geometry":@{
+                     @"type":@"GeometryCollection",
+                     @"geometries":[self geometryCollection:self.shapes]
+                     }
+             };
 }
 
+- (NS_MUTABLE_ARRAY_OF(NS_DICTIONARY_OF(NSString *, id) *) *)geometryCollection:(NS_ARRAY_OF(MGLShape *) *)shapes {
```

Comment:
MGLShapeCollectionFeature already knows about its shapes, so there’s no need to pass it in here. Just use `self.shapes` inside the method.